base/src/main/java/org/aya/resolve/context/ModuleExport.java

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) 2020-2022 Tesla (Yinsen) Zhang.
+// Use of this source code is governed by the MIT license that can be found in the LICENSE.md file.
+package org.aya.resolve.context;
+
+import kala.collection.Map;
+import kala.collection.SeqView;
+import kala.collection.SetView;
+import kala.collection.immutable.ImmutableSeq;
+import kala.collection.mutable.MutableList;
+import kala.collection.mutable.MutableMap;
+import kala.control.Option;
+import kala.tuple.Tuple;
+import kala.tuple.Tuple2;
+import org.aya.ref.AnyVar;
+import org.aya.resolve.error.NameProblem;
+import org.aya.util.error.SourcePos;
+import org.aya.util.error.WithPos;
+import org.aya.util.reporter.Problem;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public record ModuleExport(@NotNull MutableMap<String, AnyVar> exports) {
+  public ModuleExport() {
+    this(MutableMap.create());
+  }
+
+  /**
+   * @return false if failed
+   */
+  public boolean export(@NotNull String name, @NotNull AnyVar ref) {
+    var exists = putIfAbsent(name, ref);
+    return exists.isEmpty();
+  }
+
+  // TODO: delete this garbage
+  @Deprecated
+  public @NotNull Result filter(@NotNull ImmutableSeq<String> names, boolean isUse, @NotNull SourcePos pos) {
```

Comment:
I hope the `Name`s in UseHide are `WithPos`.

Reply:
These names are also serialized, so they do not always have `SourcePos`. Is there another way around?

Reply:
But done.